xbmc/threads/SystemClock.h

```diff
@@ -32,36 +34,39 @@ template<typename T>
 class EndTime<T, true>
 {
 public:
-  explicit EndTime(const T duration)
-    : m_startTime(std::chrono::steady_clock::now()), m_totalWaitTime(duration)
-  {
-  }
+  explicit EndTime(const T duration) { Set(duration); }
 
   EndTime() = default;
   EndTime(const EndTime& right) = delete;
   ~EndTime() = default;
 
+  static constexpr T Max() { return m_max; }
+
   void Set(const T duration)
   {
     m_startTime = std::chrono::steady_clock::now();
-    m_totalWaitTime = duration;
+
+    if (duration > m_max)
+    {
+      m_totalWaitTime = m_max;
+      CLog::Log(LOGDEBUG, "duration ({}) greater than max ({}) - duration will be truncated!",
```

Comment:
Better change to LOGWARNING to become always visible

Reply:
updated